test_conformance/extensions/cl_khr_command_buffer/cl_khr_command_buffer_mutable_dispatch/mutable_command_arguments.cpp

```diff
@@ -572,49 +608,60 @@ struct MutableDispatchNullArguments : public BasicMutableCommandBufferTest
                                           nullptr, nullptr);
         test_error(error, "clEnqueueCommandBufferKHR failed");
 
-        // Check the results of the modified execution
+        error = clFinish(queue);
+        test_error(error, "clFinish failed");
```

Comment:
To make sure I understand this change correctly, we are inserting a call to `clFinish()` on an in-order queue, between
a call to enqueuing the command-buffer and a blocking `clEnqueueReadBuffer` on the same queue.

Although this isn't an invalid change to the test, I don't think the previous code was incorrect. Due to the in-order queue property the `clEnqueueReadBuffer` should have a dependency on the `clEnqueueCommandBufferKHR` completing.

If this doesn't work with OCK, then I feel it's a bug with OCK that should be reported there rather than weakening the test.

Reply:
Indeed, that was the motivation behind that correction so I restored previous approach, BTW shouldn't we then remove then line [744](https://github.com/shajder/OpenCL-CTS/blob/490efa8ddbed9f98c19ef90018f20953c0410829/test_conformance/extensions/cl_khr_command_buffer/cl_khr_command_buffer_mutable_dispatch/mutable_command_arguments.cpp#L744) and [767](https://github.com/shajder/OpenCL-CTS/blob/490efa8ddbed9f98c19ef90018f20953c0410829/test_conformance/extensions/cl_khr_command_buffer/cl_khr_command_buffer_mutable_dispatch/mutable_command_arguments.cpp#L767) as well ? It is followed by blocking `clEnqueueSVMMap`, right ?

Reply:
Yes, I think we could remove those lines you linked as well.